File: claude-code/configs/harness/agent-sdk-example.py
"""
agent_sdk_example.py - Claude Agent SDK Python example
Run: pip install claude-agent-sdk && python agent_sdk_example.py
"""

import asyncio
import os
from claude_agent_sdk import query, ClaudeAgentOptions


async def run_agent(
    prompt: str,
    working_directory: str = ".",
    model: str = "sonnet",
    max_turns: int = 20,
    allowed_tools: list[str] | None = None,
):
    """Run a Claude Code agent with the given configuration."""
    if allowed_tools is None:
        allowed_tools = ["Read", "Edit", "Bash", "Glob", "Grep", "Write"]

    result = ""
    async for message in query(
        prompt=prompt,
        options=ClaudeAgentOptions(
            working_directory=working_directory,
            allowed_tools=allowed_tools,
            model=model,
            max_turns=max_turns,
            permission_mode="bypassPermissions",
        ),
    ):
        if hasattr(message, "result"):
            result = message.result

    return result


async def main():
    repo_path = os.environ.get("REPO_PATH", ".")
    task = os.environ.get("TASK", "Summarize this project")

    print(f"🤖 Running agent on: {repo_path}")
    print(f"   Task: {task}")
    print("---")

    result = await run_agent(prompt=task, working_directory=repo_path)
    print(result)


if __name__ == "__main__":
    asyncio.run(main())

File: claude-code/configs/harness/agent-sdk-example.ts
/**
 * agent-sdk-example.ts - Claude Agent SDK TypeScript example
 * Run: npm install @anthropic-ai/claude-agent-sdk && npx tsx agent-sdk-example.ts
 */

import { query } from "@anthropic-ai/claude-agent-sdk";

async function runAgent(options: {
  prompt: string;
  workingDirectory?: string;
  model?: string;
  maxTurns?: number;
  allowedTools?: string[];
}): Promise<string> {
  const {
    prompt,
    workingDirectory = ".",
    model = "sonnet",
    maxTurns = 20,
    allowedTools = ["Read", "Edit", "Bash", "Glob", "Grep", "Write"],
  } = options;

  let result = "";

  for await (const message of query({
    prompt,
    options: {
      workingDirectory,
      allowedTools,
      model,
      maxTurns,
      permissionMode: "bypassPermissions",
    },
  })) {
    if ((message as any).result) {
      result = (message as any).result;
    }
  }

  return result;
}

async function main() {
  const repoPath = process.env.REPO_PATH || ".";
  const task = process.env.TASK || "Summarize this project";

  console.log(`🤖 Running agent on: ${repoPath}`);
  console.log(`   Task: ${task}`);
  console.log("---");

  const result = await runAgent({
    prompt: task,
    workingDirectory: repoPath,
  });

  console.log(result);
}

main().catch(console.error);
